src/common/metrics.rs

```diff
@@ -174,6 +176,68 @@ impl MetricsProvider for CollectionsTelemetry {
             MetricType::GAUGE,
             vec![gauge(vector_count as f64, &[])],
         ));
+
+        let mut optimizer_counter_metrics = vec![];
+
+        let mut total_optimizations_running = 0;
+
+        for collection in self.collections.iter().flatten() {
+            let collection = match collection {
+                CollectionTelemetryEnum::Full(collection_telemetry) => collection_telemetry,
+                CollectionTelemetryEnum::Aggregated(_) => {
+                    continue;
+                }
+            };
+
+            total_optimizations_running += collection
+                .shards
+                .iter()
+                .flatten()
+                .filter_map(|i| i.local.as_ref())
+                .flat_map(|i| i.optimizations.log.iter())
+                .flatten()
+                .filter(|i| i.status == TrackerStatus::Optimizing)
+                .count();
+
+            // Sum the optimization triggers over all shards of this collection.
+            let optimizer_triggers = collection
+                .shards
+                .iter()
+                .flatten()
+                .filter_map(|i| i.local.as_ref())
+                .map(|i| i.optimizations.triggers)
+                .fold(OptimizerTriggers::default(), |total, state| total + state);
+
+            let mut add_optimizer_metric = |name: &str, value: usize| {
+                if value > 0 {
+                    optimizer_counter_metrics.push(counter(
+                        value as f64,
+                        &[("id", &collection.id), ("optimizer", name)],
+                    ));
+                }
+            };
+
+            add_optimizer_metric("config-mismatch", optimizer_triggers.config_mismatch);
+            add_optimizer_metric("index", optimizer_triggers.index);
+            add_optimizer_metric("merge", optimizer_triggers.merge);
+            add_optimizer_metric("vacuum", optimizer_triggers.vacuum);
```

Comment:
Hit: let's destruct the struct here, so that we don't forget any new fields we might add in the future.